pkg/provider/env_local.go

```
@@ -185,15 +185,35 @@ func (e *localEnvironment) ProviderConfigs() []analyzerprovider.Config {
 	rules := make([]string, len(userRules))
 	copy(rules, userRules)
-	if enableDefaults {
-		rulesetsPath := filepath.Join(e.cfg.KantraDir, rulesetsSubdir)
-		rules = append(rules, rulesetsPath)
+	if !enableDefaults {
+		return rules, nil
 	}
+	rulesetsRoot := filepath.Join(e.cfg.KantraDir, rulesetsSubdir)
+	if e.cfg.ExternalOnly {
+		if st, err := os.Stat(rulesetsRoot); err == nil && st.IsDir() {
+			rules = append(rules, rulesetsRoot)
```

golangci-lint (2.12.1)

[error] append to slice `rules` with non-zero initialized length

(makezero)